fastly_slurper/slurper.py

```diff
@@ -52,65 +47,96 @@ def request(self, method, url, *args, **kwargs):
 
 
 class RecorderWorker(threading.Thread):
+    daemon = True
 
     def __init__(self, client, publisher, service, delay=1.0):
         super(RecorderWorker, self).__init__()
-        self.daemon = True
-
         self.client = client
         self.publisher = publisher
         self.name, self.channel = service
         self.delay = delay
 
-    def timing(self, stat, time):
-        stat = '%s.%s' % (self.name, stat)
-        return self.publisher.timing(stat, time)
-
-    def gauge(self, stat, time):
-        stat = '%s.%s' % (self.name, stat)
-        return self.publisher.gauge(stat, time)
-
     def url_for_timestamp(self, ts):
-        # Convert timestamp to str and remove the decimal
-        strts = ('%.9f' % ts).translate(None, '.')
-        return join('channel', self.channel, 'ts', strts)
-
-    def get_stats(self, ts):
-        response = self.client.get(self.url_for_timestamp(ts)).json()
-        return response['Data']
-
-    def record_stats(self, message):
-        for stats in message:
-            if 'datacenter' in stats:
-                for dc, dcstats in six.iteritems(stats['datacenter']):
-                    for stat, val in six.iteritems(dcstats):
-                        if stat == 'miss_histogram':
-                            continue
-
-                        if stat.endswith('_time'):
-                            t = stat.split('_')[0]
-                            if dcstats[t]:
-                                val = val / dcstats[t] * 1000
-
-                        stat_name = '%s.%s' % (dc, stat)
-                        self.timing(stat_name, val)
-
-                self.gauge('last_record', int(time()))
-
-    def run(self):
-        self.running = True
-
-        while self.running:
-            ts = time()
+        strts = '%.9f' % ts
+        strts = strts.replace('.', '')
+        return '/'.join(['channel', self.channel, 'ts', strts])
+
+    def fetch(self, ts):
+        url = self.url_for_timestamp(ts)
+        return self.client.get(url)
+
+    def unwrap_resp(self, resp):
+        # data data data
+        data = resp.json()
+        data = data['Data']
+        # data
+        return data
+
+    def parse(self, iterable, skip_keys=['miss_histogram']):
+        for stats in iterable:
+            dc = stats.get('datacenter')
+            if not dc:
+                continue
+
+            for dc_name, dc_stats in six.iteritems(dc):
+                for k, v in six.iteritems(dc_stats):
+                    if k in skip_keys:
+                        continue
+
+                    if k.endswith('_time'):
+                        kpart = k.rsplit('_', 1)[0]
+
+                        # TODO Wat
+                        if dc_stats.get(kpart):
+                            v = v / dc_stats[kpart] * 1000
+
+                    k = '%s.%s' % (dc_name, k)
+                    yield k, v
```

Comment:
This whole for loop makes no sense to me :D May be some comments explaining what we are doing?